src/main/java/uk/gov/hmcts/ccd/domain/service/getevents/AuthorisedGetEventsOperation.java

```diff
@@ -28,30 +31,38 @@
     private final AccessControlService accessControlService;
     private final CaseDefinitionRepository caseDefinitionRepository;
     private final UserRepository userRepository;
+    private final CaseUserRepository caseUserRepository;
 
     public AuthorisedGetEventsOperation(@Qualifier("classified") GetEventsOperation getEventsOperation,
                                         @Qualifier(CachedCaseDefinitionRepository.QUALIFIER)
                                             CaseDefinitionRepository caseDefinitionRepository,
                                         AccessControlService accessControlService,
-                                        @Qualifier(CachedUserRepository.QUALIFIER) UserRepository userRepository) {
+                                        @Qualifier(CachedUserRepository.QUALIFIER) UserRepository userRepository,
+                                        CaseUserRepository caseUserRepository) {
 
         this.getEventsOperation = getEventsOperation;
         this.accessControlService = accessControlService;
         this.caseDefinitionRepository = caseDefinitionRepository;
         this.userRepository = userRepository;
+        this.caseUserRepository = caseUserRepository;
     }
 
     @Override
     public List<AuditEvent> getEvents(CaseDetails caseDetails) {
 
         final List<AuditEvent> events = getEventsOperation.getEvents(caseDetails);
 
-        return secureEvents(caseDetails.getCaseTypeId(), events);
+
+        return secureEvents(caseDetails.getCaseTypeId(), getUserRoles(caseDetails.getId()), events);
     }
 
     @Override
     public List<AuditEvent> getEvents(String jurisdiction, String caseTypeId, String caseReference) {
-        return secureEvents(caseTypeId, getEventsOperation.getEvents(jurisdiction, caseTypeId, caseReference));
+        List<AuditEvent> auditEvents = getEventsOperation.getEvents(jurisdiction, caseTypeId, caseReference);
+        Set<String> caseUserRoles = auditEvents != null && auditEvents.size() > 0
```

Comment:
I think I preferred the previous version of the code where the access roles where retrieved inside the `secureEvents` itself. Much less duplication. 1 call to `getUserRoles` vs 3 now. And if we do that we could remove lines 62 -> 64 that seem a bit of a complication to me.

Reply:
We have to pass caseId to the secureEvents... method to get the caseroles.. is that fine....
Still we have to do lines 62-64 to get caseid for that method